src/Config/ConfigFileWatcher.cs

```diff
@@ -39,16 +40,24 @@ public class ConfigFileWatcher
     /// </summary>
     public event EventHandler? NewFileContentsDetected;
 
-    public ConfigFileWatcher(string directoryName, string configFileName)
-    {
-        _fileWatcher = new FileSystemWatcher(directoryName)
-        {
-            Filter = configFileName,
-            EnableRaisingEvents = true
-        };
+    /// <summary>
+    /// Path of the directory being watched by the file watcher.
+    /// </summary>
+    public string WatchedDirectory { get; private set; }
 
-        _runtimeConfigHash = FileUtilities.ComputeHash(filePath: directoryName + "/" + configFileName);
+    /// <summary>
+    /// Name of the file being watched by the file watcher.
+    /// </summary>
+    public string WatchedFile { get; private set; }
+
+    public ConfigFileWatcher(IFileSystemWatcher fileWatcher, string directoryName, string configFileName)
+    {
+        WatchedDirectory = directoryName;
+        WatchedFile = configFileName;
+        _fileWatcher = fileWatcher;
+        _fileWatcher.Path = WatchedDirectory;
         _fileWatcher.Changed += OnConfigFileChange;
+        _runtimeConfigHash = FileUtilities.ComputeHash(_fileWatcher.FileSystem, filePath: WatchedDirectory + "/" + WatchedFile);
```

Comment:
should we utilize any cross platform file utility helper (if dotnet provides one) to specify the path separator? For example, on linux it is / but on windows it would be `\` ?
Perhaps use Path.IO.combine to get the full path name

Reply:
updated.